enhancements/cluster-logging/cluster-logging-kibana-multitenancy.md

```diff
@@ -0,0 +1,188 @@
+obsolete.
+
+## Motivation
+
+Currently the openshift-elasticsearch-plugin intercepts calls from Kibana, rewrites them
+based on the username to match up with an user-specific index, then rewrites the response so that Kibana does not detect that things have been rewritten. This currently requires that we are able to
+keep up with any changes to the Kibana -> Elasticsearch API for intercepting and updating
+the calls, this instead would be handled by Kibana before sending a request to Elasticsearch.
+
+### Goals
+The specific goals of this proposal are:
+
+* Provide a means to link OCP users to Kibana tenants and provide Elasticsearch log access
+based on their OCP access.
+
+We will be successful when:
+
+* An OCP user is able to log into Kibana and have a user specific tenant provided for them
+* Their tenant has the appropriate index patterns assigned to them
+* They should be able to share visualizations between accounts
+
+### Non-Goals
+
+* This is to configure Kibana such that users have their user tenants created for them and the appropriate
+index patterns are made available. Role access is separate from this implementation and will only be tying
+an index pattern to a specific user's role.
+
+## Proposal
+
+The OpenDistro Multitenancy plugin seeks to simplify our current process of catching requests from Kibana before they are processed by elasticsearch where we then rewrite the index to uniquely match an user. The tenant patterns that are matched to
+an user are defined within the user roles that are seeded into Elasticsearch. This functionality is to be provided by the Elasticsearch Proxy.
+
+### User Stories
+
+#### As an OKD admin, I want to be able to view all logs that I have access to
+
+This is the current state, admin users should still be able to view all the cluster logs with the provided index patterns
+
+#### As an OKD user, I want to be able to view all logs that I have access to
+
+This is current state, however instead of an index pattern per namespace the user has access to, it will be a single index pattern to match the rollover data design.
+
+#### As an OKD user, I want to be able to create and share visualizations for logs
+
+Currently users are unable to share their visualizations since Kibana did not have knowledge on how to access the specific user's Kibana index where the visualizations were stored. They would be able to share them now as Kibana specifies the tenant as part of the visualization path when sharing.
+
+### Implementation Details
+
+#### Assumptions
+
+* The Elasticsearch Proxy will configure roles for users such that they have a tenant pattern defined that matches what Kibana would create for an user logging in.
+* Each user would have only two tenants available to them: Global and their user name based tenant
```

Comment:
Initially if we are able to restrict the 'Global' tenant then we should consider doing so.  I'm not sure of the value or how users of namespace A can share visualizations of namespace B without needing access to both.

Reply:
so then you're suggesting that we only have the user tenants?

Reply:
i need to look into whether or not we can define tenants for users in addition to the ones provided by the kibana plugin

Reply:
@jcantrill we are able to create tenants, they can be seeded similar to `roles`
https://opendistro.github.io/for-elasticsearch-docs/docs/security-access-control/multi-tenancy/#add-tenants

does this cover the `admin` case that we talked about on slack?